Project-Days-Calendar/README.md

```diff
@@ -0,0 +1,111 @@
+
+Regardless of your group size, you must:
+
+* Create an HTML page which, when loaded, displays a calendar.
+* The calendar must show every day of the current month, each as a rectangle. Each row of rectangles must show one week. The first column must show Mondays. The first day of the month must be shown in the first row.
+* There must be two links or buttons which, when clicked, switch what is displayed. One button must change the display to the previous month. The other button must change the display to the next month. On repeated clicks, these buttons must keep moving back/forwards in time, one month per click.
+* There must be a way to jump to a particular month and year, e.g. "October 2020".
+* The days from the JSON file must appear correctly when the month they fall in is displayed. For example:
+  * If October 2024 is being shown, October 8th must show Ada Lovelace Day.
+  * If October 2025 is being shown, October 14th must show Ada Lovelace Day.
+* The calendar should work for every year - if someone goes to 1900, or 2050, or any other year, the commemerative days should be correctly displayed.
+* The calendar must be dynamically generated by reading the JSON file. You must not hard-code any days in the HTML file.
+* Other than the above styling requirements, no styling is required.
+
+### Requirements for groups of at least 2
+
+As well as all previous requirements, you must:
+
+* Produce a script that can be run in a terminal via `node`, which outputs to standard out (via calls to `console.log`) an [iCal format](https://icalendar.org/) file containing entries for every day in the JSON file. You must not use recurring events. There must be one entry per commemerative day per year from 2020 until 2030 (inclusive).
+* If you save this output to a file, and import it into Google Calendar, it should show the days correctly in the calendar. There is a suggested workflow for testing this listed below.
+* Logic for calculating dates must be shared between the web generator and the iCal generator.
+
+### Requirements for groups of at least 3
+
+As well as all previous requirements, you must:
+* Fetch a description of the day from the URL supplied in the JSON file.
+* In the HTML page, if you click on a listed commemorative day, the page should show a description of the day.
+* In the iCal file, the **DESCRIPTION** field should contain the description of the day.
+
+## How to test Google Calendar imports
+
+Go to https://calendar.google.com
+
+We recommend you create a new empty calendar for testing. To do this:
+1. In the bar on the left next to "Other Calendars", click the "+" and then "Create new calendar".
+2. Give your calendar a name like "Test calendar".
+
+Import your `.ics` file into the calendar. To do this:
+1. In the bar on the left next to "Other Calendars", click the "+" and then "Import".
+2. Select your test calendar from the "Add to calendar" drop down".
+3. Select the `.ics` file you have created.
+4. Press "Import".
+
```

Comment:
I wonder if it's worth including some instructions on wiping all events in a calendar so that it's easy to debug? (I actually don't know if this is possible - might be simplest to just delete the whole calendar an re-create)